### ISSUE TITLE
Vote about the LICENSE of ZeroNet

### ISSUE THREAD

**Issue by antifa** [antifa/ZeroNet#1]
# Vote about the LICENSE of ZeroNet

I created a new poll for voting about the LICENSE of ZeroNet because the [old poll](https://github.com/HelloZeroNet/ZeroNet/issues/2273) is limited to me, I'm excluded from voting while I have collaborated to ZeroNet core, I was also who reported the problem with the `LICENSE` and my issues (https://github.com/HelloZeroNet/ZeroNet/issues/2265), (https://github.com/HelloZeroNet/ZeroNet/issues/2268) and the pull request (https://github.com/HelloZeroNet/ZeroNet/pull/2270) what I opened with the new **GPLv3+** `LICENSE` is censored, closed and discredited.

The poll which created by @krixano and @imachug intentionally misleading all previous ZeroNet collaborators and so I believe that is in the best interest of every ZeroNet collaborators if I start a new poll. 

## About the poll options: 

There is only two options in this poll, either **GPLv3+** or **GPLv3**. 

The **GNU General Public License v3.0** (GPLv3) **is compatible** with all other licenses, namely with AGPL 3.0 or later (agpl-3.0-plus), GPL-3.0-only (gpl-3.0), GPL-1.0-or-later (gpl-1.0-plus), BSD-2-Clause (bsd-simplified), BSD-3-Clause (bsd-new), MIT (mit) and Apache 2.0 (apache-2.0) all of which is included in the current ZeroNet source code.

<img width="733" alt="compatibility-chart" src="https://user-images.githubusercontent.com/57236964/68153477-7440c300-ff3d-11e9-9ed6-3c0e7c55a310.png">

#### Why the voting is necessary? 

Because the current license of ZeroNet namely GPLv2 (GPL-2.0-only) is for years violated other peoples copyrights and licenses and it is incompatible with most licenses ZeroNet has in the source code. 

It is **not permitted** to switch from GPL to any other license for example to Apache 2.0 because it is not possible to revoke the permissions what is already granted to those who forked ZeroNet or obtained the source code of ZeroNet with a GPL license.

### Poll:
Vote by clicking on one of the options below! 

[![](https://api.gh-polls.com/poll/01DRVZBXYQ0BHM8G9DGJXDYJ3T/GPLv3%2B)](https://api.gh-polls.com/poll/01DRVZBXYQ0BHM8G9DGJXDYJ3T/GPLv3%2B/vote)
[![](https://api.gh-polls.com/poll/01DRVZBXYQ0BHM8G9DGJXDYJ3T/GPLv3)](https://api.gh-polls.com/poll/01DRVZBXYQ0BHM8G9DGJXDYJ3T/GPLv3/vote)

Thank you for your vote!

**Comment by antifa** [comment]
@shortcutme @HelloZeroNet @imachug @rllola @tangdou1 @TheNain38 @jerry-wolf @radfish @MuxZeroNet @matthewrobertbell @grez911 @sirMackk @Idealcoder @rainlime @ysc3839 @barrabinfc @0polar @filips123 @cclauss @DaniellMesquita @anoadragon453 @n3r0-ch @OliverCole @Th3B3st @geekless @cxgreat2014 @erqan @iShift @mkg20001 @krzotr @krixano @nathantym @Emeraude @frerepoulet @aitorpazos @jTeego @yowmamasita @reezer @danielquinn @HostFat @JeremyRand @volker48 @tormath1 @rarbg @ppsfassa @brunogarciavaz @caryoscelus @hugbubby @mishfit @vitorio @xfq @6543 @ajmeese7 @AceLewis @megfault @zasei @artemmolotov @Nephos @Austin-Williams @bencevans @valkheim @d14na @thesoftwarejedi @Derson5 @dldx @EdenSG @camponez @Erkan-Yilmaz @Fil @gyulaweber @shakna-israel @flibustier @justinwiley @kseistrup @MRoci @sexybiggetje @BoboTiG @medimatrix @Nodeswitch @Ornataweaver @adrelanos @quasiyoke @Radtoo @RedbHawk @rcmorano @rubo77 @SuperSandro2000 @Thunder33345 @anonym @beigexperience @blurHY @dqwyy @eduaddad @goofy-mdn @krikmo @leycec @mnlg @mymage @probonopd @saber28 @rwv @sinkuu @zwgshr

**Comment by antifa** [comment]
@xfq @6543 @ajmeese7 @AceLewis @megfault @zasei @artemmolotov @Nephos @Austin-Williams @bencevans @valkheim @d14na @thesoftwarejedi @Derson5 @dldx @EdenSG @camponez @Erkan-Yilmaz @Fil @gyulaweber @shakna-israel @flibustier @justinwiley @kseistrup @MRoci @sexybiggetje @BoboTiG @medimatrix @Nodeswitch @Ornataweaver @adrelanos @quasiyoke @Radtoo @RedbHawk @rcmorano @rubo77 @SuperSandro2000 @Thunder33345 @anonym @beigexperience @blurHY @dqwyy @eduaddad @goofy-mdn @krikmo @leycec @mnlg @mymage @probonopd @saber28 @rwv @sinkuu @zwgshr

**Comment by antifa** [comment]
@rwv @sinkuu @zwgshr

**Comment by mishfit** [comment]
Stop tagging us, blocked.

**Comment by imachug** [comment]
> It is not permitted to switch from GPL to any other license for example to Apache 2.0 because it is not possible to revoke the permissions what is already granted to those who forked ZeroNet or obtained the source code of ZeroNet with a GPL license

So *that's* what you didn't like... Of course we'll distribute old ZeroNet versions under GPLv2, but all following releases will be licensed under another license.

**Comment by caryoscelus** [comment]
> It is not permitted to switch from GPL to any other license for example to Apache 2.0 because it is not possible to revoke the permissions what is already granted to those who forked ZeroNet or obtained the source code of ZeroNet with a GPL license.

Indeed, nobody can **revoke** GPL license to the **existing code** it covers. What you're missing is that if all "copyright holders" agree, it's possible to license the code under a new license, and if they so desire, the **new contributions** will be available **only** under that license.

Please, try to investigate the issue before spreading FUD and sparking hostility

**Comment by filips123** [comment]
> It is not permitted to switch from GPL to any other license for example to Apache 2.0 because it is not possible to revoke the permissions what is already granted to those who forked ZeroNet or obtained the source code of ZeroNet with a GPL license.

Again. Future releases can change license. In this case, nobody is revoking old GPL license and you can still use old releases as GPL.

**Comment by antifa** [comment]
@caryoscelus I understand you, but the new contributions can't limit the users who still have GPL licensed source code. You can't impose new restrictions on the GPL covered works at all. For example the released GPL code can't be reused under the new license! 

@imachug You can't distribute under GPLv2 because that is and all other license in that version of the source code is automatically terminated according to the terms of those licenses.

@filips123 In that case is the best if nobody contributes to the repository of @HelloZeroNet. **The old license is terminated** and all other too. There is nothing you can do to restore that license except removing the conflicting code and licenses.

**Comment by filips123** [comment]
> The old license is terminated and all other too. There is nothing you can do to restore that license except removing the conflicting code and licenses.

If the old license is terminated, there is also no obligality to still use GPL license.

**Comment by antifa** [comment]
Well, no obligation for who? I received under GPL and I'm obligated to comply with all licenses which the source code includes, that is why I upgraded the license and notified everyone in the repository of @HelloZeroNet about this issue.

**Comment by filips123** [comment]
Old versions will remain either GPL licensed (if the license was actually not revoked), proprietary or public domain. So in all cases, the only thing you would need to do to "comply with all licenses" would be to update ZeroNet to new version which would use new license.

Also, who has more rights? One `LICENSE` file or actual contributor who made the code?

**Comment by mkg20001** [comment]
@antifa Per law a copyright holder may grant anyone permission for usage of code. Per the LICENSE in the repo, all added code can automatically be used under that license and that license may or may not be revocable under it's terms. Yet every copyright holder can choose to grant permission using _another_ license since they still _own_ the rights to do that, unless they've waived them using the LICENSE. Which they didn't.

**Comment by caryoscelus** [comment]
> You can't impose new restrictions on the GPL covered works at all

Ever head of dual licensing? Or that copyright holders have no obligation to follow the license because they have rights to the work obtained in another way?

Again, please research the issue before claiming incorrect things. Read copyright law, or any faq about dual licensing, or any story about authors closing down previously open-source works

**Comment by antifa** [comment]
@mkg20001 I do have copyright in ZeroNet just like you or anyone else. So @filips123 I have the same rights as you whatever you given me credit or not!

**Comment by filips123** [comment]
For everyone mentioned in this issue:

Sorry about that. This is not an official poll about ZeroNet license. The official poll is in HelloZeroNet/ZeroNet#2273. Please use that issue for any (civilized) discussion or voting for ZeroNet license.

Thank you!

**Comment by antifa** [comment]
As I said it, it is locked and I'm entirely excluded from that repository while I did fixed various issues and without my report this voting there would be not even exist.

**Comment by filips123** [comment]
> You can't impose new restrictions on the GPL covered works at all

So do you think Lax license (like MIT or BSD) has more restrictions than GPL?

**Comment by antifa** [comment]
@filips123 I did not said that! However, ZeroNet in order to be compatible with those licenses must upgrade to GPLv3 preferably GPLv3+.

**Comment by filips123** [comment]
> without my report this voting there would be not even exist.

Without your report, nobody would care about that "license violations" and everything would be fine. I'm mostly sure that maintainers of projects whose "license has been violated" also won't have any problems with this.

**Comment by antifa** [comment]
**I do have problem with ZeroNet if for years violating other peoples copyright and licenses and than even violates my copyrights and giving me no credit for my contributions!**

**Comment by mkg20001** [comment]
> I do have problem with ZeroNet if for years violating other peoples copyright and licenses and than even violates my copyrights and giving me no credit for my contributions!

DMCA them. But don't bother if they sue you for spam or just remove your code.

**Comment by antifa** [comment]
@mkg20001 DMCA is a waste of time. I was gently and respectfully reported the problem with the license and I was immediately attacked in the repository of @HelloZeroNet.

**Comment by antifa** [comment]
How dare I'm that I bringing to light copyright infringements and license violations. Right. I'm the evil!

**Comment by rarbg** [comment]
Someone said copyright so now you got my interest :D

**Comment by filips123** [comment]
@antifa If you haven't seen yet, ZeroNet has organized pool/voting to decide which license would be used in the future. So do you think this is not enough? Do you expect them to collect all contributor consents in 1 hour and release a fix right after that?

**Comment by antifa** [comment]
ZeroNet not organized that vote! @imachug and @krixano did! Both of them wanted to restrict the licenses but they slowly realized I'm right. I was excluded from the beginning from the vote which is why I need to open various issues and reference the vote.

**Comment by filips123** [comment]
> Both of them wanted to restrict the licenses but they slowly realized I'm right.

Releasing as MIT or BSD would make ZeroNet less restricted.

**Comment by antifa** [comment]
@filips123 I agree with you completely but the problem is that ZeroNet than need to remove many code from its source code and you can't force more than 1,900 people who forked (and who knows how many downloaded) to follow the new license or delete some code from they source code. 

The best and easiest is to upgrade to GPLv3+ and all problem solved. ZeroNet can keep all code and licenses which is included in the source code and other people like myself will be free to do whatever we wish! Even including ZeroNet in a patent claim without obtaining exclusive rights to the software!

**Comment by antifa** [comment]
@filips123 @imachug @rarbg @caryoscelus 

**GPLv3: Better Copyleft for Developers and Users**
<a href="http://www.youtube.com/watch?feature=player_embedded&v=nfLcUccWSco" target="_blank"><img src="http://img.youtube.com/vi/nfLcUccWSco/0.jpg" alt="GPLv3: Better Copyleft for Developers and Users" width="858" height="480" border="10" /></a>

:+1: :clap: :clap: I'm so sad that less than 200 people seen this! :sob:

**Comment by filips123** [comment]
This would be how a process of resolving licenses and modularization of ZeroNet could/would probably look like if "GPLv3 and Lax" is chosen:

1. First, ZeroNet is just re-licensed to GPLv3 to resolve all currently incompatibilities.

2. Then, we (here I mean ZeroNet maintainers and contributions) start working on modularization.

3. During that, new contributions are asked (while making their PR) for simple CLA to give as ability to re-license their work as MIT/Lax license if their work would be later used on ZeroNet core libraries.

4. ZeroNet is then modularized into multiple libraries/packages. Core libraries that can be used for building third-party Zeronet-based solutions are licensed under MIT/Lax license which gives developers ability to use them easily. Main ZeroNet program (as whole like it is currently) remains licensed under GPLv3 as it is not meant to be frequently reused in third-party solutions.

5. In case if there are some libraries which needs to be used in MIT-licensed part of ZeroNet, we try to resolve this. In worst case, we find alternative libraries which are also compatible with MIT.

This process could take quite a long time but it would make ZeroNet actually bless restricted.

**Comment by antifa** [comment]
@filips123 I will not use modularization and I think is a bad idea. I will keep this source code this way only make it better. If you guys go for modular I will develop ZeroNet and encourage others to keep developing this version not the modular one. Far as I see in the voting GPLv3+ got the most vote. 

I see this modularization idea as an abuse of GPLv3 license because this way you try to limit the use of ZeroNet. You can't require in any way users to use a software which is not licensed under GLPv3 in order to make the software functional. This is clearly a restriction and I will not comply with anything like this.

**Comment by antifa** [comment]
@filips123 Regarding PR, I note than I'm excluded from that repository and I I can't contribute anything to that repository. I don't think we need a central authority who can decide what and how included in ZeroNet.

My pull requests was closed and many of my code was copied modified and included in ZeroNet core **without giving me any credit!** This is far as I concern a copyright infringement! Maybe @rarbg can talk about this. 

Right after that my code is stolen basically and merged to ZeroNet core I was banned from the repository which is absurd! How you imagine @filips123 this contribution process in the future? Will @shortcutme keep this practice in place? If so, than no matter what license you will use he will violate all licenses and copyright of others. 

I even thinking to rewrite ZeroNet from top to bottom. For me alone would take quite a long time but I'm not afraid to do it! 

If everyone just send code to the hello repository and don't get credit than I see no credit to have any community at all.

**Comment by krixano** [comment]
I'd like to point everyone here to this issue, where you can see the consistent abuse of @antifa and @hacktivist (his previous 4 accounts) on multiple people including @HelloZeroNet, @imachug , me, and @filips123 . Don't let this person fool you @rarbg . @antifa tried to implement a crazy caching feature into ZeroNet and  all of us, including @HelloZeroNet, disagreed with it, so he started all of this *out of spite.*

Regarding the copyright issue, you don't own ideas (Btw, @antifa tried saying he *does*, which is yet *another* reason not to trust him). Nobody owns the classic "Hello World" program. You can't say that @HelloZeroNet copied your code because you have no proof he *actually* copied your code. Having similar lines of code is **not** the same as copying the code. I also urge people to look into this themselves instead of take this person's word for it, because as you can see he's clearly lying about a lot of things.

And if you can't see how hypocritical and disingenuous this person is, then just look at how he's using Copyright against people but won't even abide by *other people's* copyright!

https://github.com/HelloZeroNet/ZeroNet/issues/2285#issuecomment-549461502

I retained all the evidence that I could take screenshots of, which is more than @antifa can say since he deliberately deleted his comments. Who's the dishonest one now?

**Comment by filips123** [comment]
> I see this modularization idea as an abuse of GPLv3 license because this way you try to limit the use of ZeroNet.

Since when switching from GPL to MIT/BSD and modularizing something is limit of the usage?

Of course ZeroNet doesn't need (and probably shouldn't be) so modularized as IPFS (as they are not actually meant to be used in same way), but some modularization would help developers and make ZeroNet less restricted.

Main software will be still lisensed as GPL, and libraries will be listened as MIT/Lax, so everything will be still open source and even less restricted actually.

**Comment by krixano** [comment]
@rarbg @caryoscelus @mkg20001 Here's a 3 way diff. Center is the original before hacktivist/antifa's PR. Left is nofish's implementation. Right is hacktivist/antifa's implementation from the PR (https://github.com/HelloZeroNet/ZeroNet/pull/2248)

![Screenshot from 2019-11-05 00-28-20](https://user-images.githubusercontent.com/4098042/68183936-600fbc80-ff63-11e9-9e9d-a738de560c57.png)
![Screenshot from 2019-11-05 00-27-28](https://user-images.githubusercontent.com/4098042/68183943-643bda00-ff63-11e9-85ee-6a1d8495bf04.png)
![Screenshot from 2019-11-05 00-27-47](https://user-images.githubusercontent.com/4098042/68183947-66059d80-ff63-11e9-88fe-49786fedd3e3.png)
![Screenshot from 2019-11-05 00-28-03](https://user-images.githubusercontent.com/4098042/68183951-67cf6100-ff63-11e9-853d-33238396e591.png)

Here's a video of me scrolling through the whole file:
https://www.youtube.com/watch?v=YaxcDekczYo

**Comment by 0x6a73** [comment]
If the GPLv2 license was violated over past few years, then why do you consider only the latest version invalid? According to your thinking, all Zeronet versions and forks should be considered public domain now.

Also, lax licenses (especially ISC/MIT/BSD) allow for inclusion in GPLv2/v3 software, so I don't see a single reason why we can't release a core project as GPLv3 licensed and relicense most other code to some lax license and release it.

BTW, if you delete this comment again, I still have it saved in a file so I can repost it.

**Comment by 0x6a73** [comment]
TL;DR since they were violating the license for years, it was already revoked the moment they started doing so. And this means that forks bear these violations too, meaning that their rights aren't revoked right now because they already were revoked in the past.

**Comment by 0x6a73** [comment]
Also, if you again delete my comments, then I would really urge everyone discussing here to switch to the original ZeroNet repo and talk there, while blocking Antifa.

**Comment by imachug** [comment]
@0x6a73 How many comments has antifa deleted already??

**Comment by 0x6a73** [comment]
@imachug He's deleted two comments in an another issue already.

**Comment by caryoscelus** [comment]
@imachug looks like github still shows a stub in place of deleted comments, so you'll notice, e.g.

> Repository owner deleted a comment from mishfit 12 hours ago

in this thread

**Comment by krixano** [comment]
@caryoscelus 

I don't see that in this thread....

**Comment by caryoscelus** [comment]
@krixano are you sure? the link is https://github.com/antifa/ZeroNet/issues/1#event-2769117217

**Comment by krixano** [comment]
Yeah, I don't see it. The link doesn't go to it either.

**Comment by filips123** [comment]
@krixano I see removed comment notice. Maybe try refreshing site or view desktop/mobile version of website.

**Comment by antifa** [comment]
@krixano I do have all the email I received from GitHub in all my accounts and clearly @shortcutme coppied my code! I kindly ask everyone @filips123 to delete my contribution from the ZeroNet repository!

**Comment by antifa** [comment]
@0x6a73 **The lost all copyright when the license is terminated** this is how GPLv2 works! 
I warning that if @HelloZeroNet not upgrade urgently the license to **GPLv3+** ZeroNet development need to be stopped! Also you can't use anymore the licenses you have violated! 

The owners of `pyelliptic` and others **need to give ZeroNet the license again because you have lost all the right to use that program**! 

@filips123 please make sure everyone get it that I requested to delete my contributions form ZeroNet core or **I take this to court**!

**Comment by 0x6a73** [comment]
@antifa repeating the same thing 20 times won't make it any more valid. Also, deleting comments instead of facing them is cowardly.

**Comment by 0x6a73** [comment]
Also, please take this to court.

**Comment by antifa** [comment]
I was also the first person who publicly stated on GitHub that I use ZeroNet as a back-end (cluster) behind a reverse proxy and now I see that some people already have problem with ZeroNet because @imachug and others intentionally **put restrictions in ZeroNet** right after my comments!

**Comment by 0x6a73** [comment]
Tell me: why won't you just take this to court?

**Comment by antifa** [comment]
I first make sure @HelloZeroNet repository is blocked!

**Comment by antifa** [comment]
**I want my contributions deleted from ZeroNet immediately right now!** 
Fix you shitty problems without my contributions!

**Comment by 0x6a73** [comment]
No one listens to you here anyway apart from me.

**Comment by antifa** [comment]
@krixano do you recognize yourself? 
![Christian Seibold](https://i.ibb.co/qdG53K1/christian-cleaned.png)
Fucking license violator and copyright infringer!

**Comment by antifa** [comment]
@krixano @imachug @filips123 All of my contributions you going to remove from ZeroNet right now as I requested! I will not contribute to copyright infringers and license violators shit! I fixed various issues with my contribution what you had for years! I was not even given credit and right to vote so fuck off! Delete my contributions from ZeroNet core! @HelloZeroNet you piece of shit! Remove all of my contributions from ZeroNet before I sue you!

**Comment by 0x6a73** [comment]
Do it or block me. Go ahead.

**Comment by 0x6a73** [comment]
Sue us or block me. Go ahead  Mr. Shithead.

**Comment by antifa** [comment]
@imachug @krixano @0x6a73 I kill you man! You fucking with me? Delete my contributions from ZeroNet or I swear I fucking kill you! 
@filips123 better you do something very fast before I take my gun and justice to my own hand!

**Comment by 0x6a73** [comment]
@antifa what do you think about a longer talk on IRC later? I believe it might be a bit better than spamming Github with our rants.

**Comment by antifa** [comment]
I'm going to kill 3 person! @krixano @imachug and @shortcutme all of which criminals and copied my code to fix they shit because they by they own was unable to do it for years! 

Right after my contribution I was banned from that repo and no credit is given to me so I was started digging and turned out that they violate other peoples copyrights and licenses for years! 
Now they trying to discredit me that I made a small contribution etc, but that small contribution fixed many years long issues! I seriously going to fucking just kill them! I had enough! 

Delete my code from ZeroNet and solve it other way or I swear you will see soon a gun in your face!

**Comment by 0x6a73** [comment]
Great. I'll be back online in 1-2 hours

**Comment by antifa** [comment]
@0x6a73 I can rent a server and install IRC if you want which allowing TOR.

**Comment by 0x6a73** [comment]
Nah, Rizon is good enough, it allows connections from TOR.

**Comment by antifa** [comment]
This people call themselves developers? Stolen copyright and license violations? Look at zeronet.io for example look at this "infrastructure", this is absolute trash! They don't even know how to setup a mail or a web server and certainly they have no clue about security. But stealing my code and discrediting me that fine! Delete my code from ZeroNet core or I fucking kill each of you do you hear me @imachug ?! You fucking license abuser and copyright infringer! 

![zeronet.io](https://user-images.githubusercontent.com/57236964/68210334-355a4e00-ffcd-11e9-843d-7e94ec4ddc08.png)

**Comment by imachug** [comment]
🤦‍♂

Please show me the code you want to delete from ZeroNet repository.

**Comment by 0x6a73** [comment]
Great. We'll write a replacement.

**Comment by 0x6a73** [comment]
What's wrong? You said that you don't want your code there, so we can remove the whole file and write it from scratch.

**Comment by imachug** [comment]
Ok, wait a second. Why can't we write our own code?

**Comment by filips123** [comment]
Python's mimetypes module is also very similar. So I think that you should sue Python project to be sure.

**Comment by 0x6a73** [comment]
Please explain how is it based on your code. I've never looked at the file, so right now I can just delete it and basing on other parts of code deduce what I should write there myself. If it's not right, please explain.

Also, I'm ready to talk with you. An IRC session on Rizon from one of my data storage servers is now running. You can join the #zeronet-drama channel right now, you can start writing now and I'll read it later, but I will be away for the next 30-40 minutes. I hope we'll find a way to resolve that problem.

**Comment by imachug** [comment]
> Because it is based on mine!

Hm, no it's not.
